File: Coding_Test_Question/kakao/q1.py
#solved
def solution(s):
    answer = 0

    nums = {"zero": 0, "one": 1, "two": 2, "three": 3, "four": 4, "five": 5, "six": 6, "seven": 7, "eight": 8,
            "nine": 9}
    words = ["zero", "one", "two", "three", "four", "five", "six", "seven", "eight", "nine"]

    for w in words:
        s=s.replace(w,str(nums[w]))


    return int(s)


s = "123"
print(solution(s))

File: ProgrammersKitJS/q1.js
function solution(d, m) {
  let answer=0;
  let curBox=1;
  for(let i=0;i<d.length;i++){
    answer++;
    if(d[i]>=curBox){
      m-=curBox;
      curBox*=2;

    }else{
      curBox=1;
    }
    if(m<=0){
      return answer;
    }
  }
  return -1;
}

let d=[2,2,4,3]	;
let m=8;
console.log(solution(d,m));